setup.py

```diff
@@ -0,0 +1,176 @@
+from setuptools.command.install import install
+from distutils.command.clean import clean
+from os import path, environ
+import os, glob
+from shutil import rmtree
+
+# access some simple stuff
+here = path.abspath(path.dirname(__file__))
+py_include_dirs=list(set([get_python_inc(i) for i in range(2)]
+                         +[get_python_lib(i,j)\
+                           for i,j in zip(range(2),range(2))]))
+
+# Read in the package information generated by the makefile.
+with open('VERSION','r') as f:
+    version = float(f.read().strip())
+
+with open('README.md','r') as f:
+    long_description = f.read()
+
+try:
+    with open('includes.txt','r') as f:
+        includes = [x.strip() for x in f.readlines()]
+    includes += py_include_dirs
+    includes = [path.realpath(x) for x in includes]
+
+    with open('flags.txt','r') as f:
+        flags = f.read().strip().split(' ')
+
+    with open('links.txt','r') as f:
+        links = [x.strip() for x in f.readlines()]
+    links = [path.realpath(x) for x in links]
+    link_args = ["-Wl,-rpath,{}".format(p) for p in links]
+
+    with open('compiler.txt','r') as f:
+        cc = f.read().strip()
+    environ["CC"] = cc
+
+    with open('libs.txt','r') as f:
+        libs = [x. strip() for x in f.readlines()]
+    libs = reduce(lambda x,y: x+y,[x.split(' ') for x in libs])
+    libs = [x.lstrip('-l') for x in libs]
+
+    with open('lib_sources.txt','r') as f:
+        lib_sources = f.read().strip().split(' ')
+except IOError:
+    raise IOError("There is missing metadata. Try running "
+                  +"'make meta' to generate it.")
+
+
+all_object_files = glob.glob(path.join(here,'*.o'))
+if len(all_object_files) == 0:
+    raise IOError("There are missing object files! Try running "
+                  +"'make' to build SimulationIO.")
+object_files = [src.replace('cpp','o') for src in lib_sources]
+
+swig_modules = ['H5','SimulationIO','RegionCalculus']
+swig_module_files = [path.join(here,"{}.py".format(m)) for m in swig_modules]
+swig_module_targets = [path.join(here,'pysimulationio',"{}.py".format(m))\
+                       for m in swig_modules]
+swig_wrap_files = (
+    [path.join(here,"{}_wrap.cpp".format(m)) for m in swig_modules]
+    +[path.join(here,"{}_wrap.cxx".format(m)) for m in swig_modules]
+    +[path.join(here,"{}_wrap.o".format(m)) for m in swig_modules])
+swig_opts = ['-c++','-Wall']
+
+# Overload the build and install commands so that swig behaves. See:
+# stackoverflow.com/questions/12491328/python-distutils-not-include-the-swig-generated-module
+# I would consider this a bug in setuptools.
+class CustomBuild(build):
+    def run(self):
+        self.run_command('build_ext')
+        # We copy the *.py files generated by swig by hand.
+        # TODO: find cleaner way to handle this
+        for i in range(len(swig_modules)):
+            if path.exists(swig_module_targets[i]):
+                os.remove(swig_module_targets[i])
```

Comment:
In this case, I disagree. I could catch an IOError, but I think it's cleaner to check if the file exists. What I don't like is copying the file manually... that's gross.